pkg/cli/cli.go

```diff
@@ -35,16 +35,21 @@ func Run(module kernel.ModuleFactory, otherModuleMaps ...map[string]kernel.Modul
 
 	ctx := appctx.WithContainer(context.Background())
 
-	k, err := kernel.New(ctx, config, logger, kernel.WithKillTimeout(settings.KillTimeout))
-	if err != nil {
-		defaultErrorHandler("can not initialize the kernel: %w", err)
+	options := []kernel.Option{
+		kernel.WithKillTimeout(settings.KillTimeout),
+		kernel.WithModuleFactory("cli", module, kernel.ModuleType(kernel.TypeEssential), kernel.ModuleStage(kernel.StageApplication)),
 	}
 
-	k.Add("cli", module, kernel.ModuleType(kernel.TypeEssential), kernel.ModuleStage(kernel.StageApplication))
 	for _, otherModuleMap := range otherModuleMaps {
 		for name, otherModule := range otherModuleMap {
-			k.Add(name, otherModule)
+			options = append(options, kernel.WithModuleFactory(name, otherModule))
 		}
 	}
+
+	k, err := kernel.BuildKernel(ctx, config, logger, options)
+	if err != nil {
+		defaultErrorHandler("can not initialize the kernel: %w", err)
```

Comment:
We don't necessarily know that the default error handler doesn't return, so we should return afterwards to avoid running a nil kernel

Reply:
added the returns :+1: